src/goto-symex/goto_symex.h

```
@@ -38,6 +38,35 @@ class namespacet;
 class side_effect_exprt;
 class typecast_exprt;
 
+/// Configuration of the symbolic execution
+struct symex_configt
+{
+  const unsigned max_depth;
+  const bool doing_path_exploration;
+  const bool allow_pointer_unsoundness;
+  bool constant_propagation;
+  bool self_loops_to_assumptions;
+  bool simplify_opt;
+  bool unwinding_assertions;
+  bool partial_loops;
+  std::string debug_level;
+
+  explicit symex_configt(const optionst &options)
```

🤔 I can never decide whether parsing (or doing any operation that can fail) in the constructor is a good or bad idea. I wonder what happens when an option is not specified.

---

I could have a non-member `symex_configt symex_config(const optionst &)` function instead. Will try that.

---

Making it a separate function only makes sense if the function can return an empty optional as an error for instance (constructors cannot return empty optionals).

---

The result of that is required for the goto_symext constructor anyway so using optional just there wouldn't solve the problem. I would suggest making the `goto_symext` constructor take a `symex_configt` instead.

---

Actually get_option functions have default for when the option is not specified (0 for int, "" for string) so we can go back to using a constructor and there is no reason for it to fail.

---

Not a fan of silent failures, but this seems to be a subject for another time.